internal/infrastructure/kubernetes/service.go

```diff
@@ -101,20 +61,29 @@ func (i *Infra) expectedService(infra *ir.Infra) *corev1.Service {
 	return svc
 }
 
-// createService creates a Service in the kube api server based on the provided infra,
-// if it doesn't exist.
-func (i *Infra) createService(ctx context.Context, infra *ir.Infra) (*corev1.Service, error) {
-	expected := i.expectedService(infra)
-	err := i.Client.Create(ctx, expected)
+// createOrUpdateService creates a Service in the kube api server based on the provided infra,
+// if it doesn't exist or updates it if it does.
+func (i *Infra) createOrUpdateService(ctx context.Context, infra *ir.Infra) error {
+	svc := i.expectedService(infra)
+	err := i.Client.Create(ctx, svc)
 	if err != nil {
 		if kerrors.IsAlreadyExists(err) {
-			return expected, nil
+			// Update service if its exists
+			if err := i.Client.Update(ctx, svc); err != nil {
+				return fmt.Errorf("failed to update service %s/%s: %w",
+					svc.Namespace, svc.Name, err)
+			}
```

Comment:
IMO we should compare expected with actual, e.g. what exists in the controller-runtime cache, and only update if managed fields, e.g. `spec`, change. I'm fine proceeding with the approach in this PR for v0.2.0 as long as a tracking issue is created for doing the comparison.